### PR TITLE
apigee linting for sharedflows

### PR DESCRIPTION
**Pull request by dwvd**
Hi,

I am creating this pull request to do linting for sharedflows. 
Please let me know if you would like me to correct anything into this.

### CODE REVIEW

**Comment by apetrosian** [comment]
@dwvd thanks, I'll review once compete previous PRs

**Comment by dwvd** [comment]
@apetrosian any update regarding this?

**Comment by apetrosian** [comment]
@dwvd sorry for the delay, we are in process of migration to the official Apigee https://github.com/apigee/

I haven't tested this feature yet, but have a quick comment. It would be great if we can avoid adding extra flags and let the tool to automatically detect either normal proxy or shared flow. I'll update you once more extensive testing will be performed. Thanks!

**Comment by dwvd** [comment]
@apetrosian I updated the pull request with the suggested changes.  Requesting you to kindly approve the changes at the earliest. Kindly let me know if any details required from my end.

**Comment by dwvd** [comment]
@apetrosian  please update regarding this.

**Comment by dwvd** [comment]
> I signed it!

**Comment by dwvd** [comment]
> I signed it!

**Comment by dwvd** [comment]
> I signed it!

**Comment by dwvd** [comment]
@apetrosian It seems https://cla.developers.google.com/ is not accepting my Contributor License Agreement (CLA).  It is accepting the changes which I updated through UI (dwvd) but the one which I checked in throw gitbash it is complaining thinking that it is from other user. (Gaurav Dwivedi)

**Comment by apetrosian** [comment]
@dwvd seems like your are using different email with gitbash config. You can double check it using following command:
`git config -l `

Make sure that you are using the proper one (with accepted contributor agreement) for all commits.
https://help.github.com/en/articles/setting-your-commit-email-address-in-git

Meanwhile, I'll review your PR and let you know when it's ready to be merged.

**Comment by dwvd2006** [comment]
I signed it!

**Comment by dwvd** [comment]
@apetrosian I checked the email in git bash cert and tried signing it with it but it seems it is still not accepting it.  Will you be able to still merge the pull request? if not then I will recreate this pull request again with the same code from one consistent id. Do you have any suggestions for me on this?

**Comment by dwvd2006** [comment]
@dwvd2006 and @dwvd are the same person. Please accept this pull request.

**Comment by dwvd** [comment]
@apetrosian Hi Artem, Any update on my pull request?

File: lib/package/plugins/checkSharedFlowBundleStructure.js [hunk 1]
```diff
@@ -0,0 +1,252 @@
+//bundleStructure.js
```
Comment (apetrosian) [inline lib/package/plugins/checkSharedFlowBundleStructure.js]: maybe consider the way to integrate sharedflow into existing bundleStructure plugin or create specific plugin for shared flow from scratch (but in this case need to update the docs with instructions how to skip boundleStructure plugin for sharedflow and wise versa)

File: lib/package/policySharedFlow.js [hunk 2]
```diff
@@ -0,0 +1,238 @@
+//PolicySharedFlow.js
```
Comment (apetrosian) [inline lib/package/policySharedFlow.js]: copied from Policy.js, as mentioned previously it is better to reuse existing one

**Comment by apetrosian** [comment]
covered in #149

**Pull request closed without merging**